src/averageAges.js

```diff
@@ -55,7 +69,19 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = onlyWithSon
+    ? people.filter(p => p.sex === 'm'
+      && people.some(m => m.name === p.mother))
+    : people.filter(p => people.some(m => m.name
+      === p.mother));
+
+  let sum = 0;
+
+  children.forEach((ch) => sum += ch.born - people.find(
+    (m) => m.name === ch.mother
+  ).born);
+
+  return sum / children.length;
```

Comment:
Again issue with sum logic made some additional function

Also, try to think about naming. Dont really think that `m.name === mother` is possible. Just because m for is a man, not a mother

and don't forget that you need to find the difference so better use `difference` or `differenceSum` instead of `sum`

I give you some better examples with `forEach` and try to make similar with `reduce`
The example:
```
let difference = 0;

children.forEach(child => {
	const { born, mother } = child;

	difference += born - people.find(
	    woman => woman.name === mother
	  ).born
  });
  
return difference / children.length;

```